plugins/modules/aci_file_remote_path.py

```diff
@@ -0,0 +1,359 @@
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        name=dict(type="str"),
+        description=dict(type="str"),
+        remote_host=dict(type="str"),
+        remote_port=dict(type="int"),
+        remote_protocol=dict(type="str", choices=["ftp", "scp", "sftp"]),
+        remote_path=dict(type="str"),
+        auth_type=dict(type="str", choices=["password", "ssh_key"]),
+        remote_user=dict(type="str"),
+        remote_password=dict(type="str", no_log=True),
+        remote_ssh_key=dict(type="str", no_log=True, aliases=["remote_key"]),
+        remote_ssh_passphrase=dict(type="str", no_log=True, aliases=["passphrase"]),
+        management_epg=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["name", "auth_type"]],
+            ["state", "absent", ["name"]],
+        ],
+        required_together=[
+            ["remote_host", "remote_port"]
+        ],
+    )
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    remote_host = module.params.get("remote_host")
+    remote_port = module.params.get("remote_port")
+    remote_protocol = module.params.get("remote_protocol")
+    remote_path = module.params.get("remote_path")
+    auth_type = module.params.get("auth_type")
+    remote_user = module.params.get("remote_user")
+    remote_password = module.params.get("remote_password")
+    remote_key = module.params.get("remote_ssh_key")
+    passphrase = module.params.get("remote_ssh_passphrase")
+    management_epg = module.params.get("management_epg")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    if auth_type == "password":
+        if remote_key is not None:
+            aci.fail_json(msg="remote_key cannot be set if auth_type is password")
+        if passphrase is not None:
+            aci.fail_json(msg="passphrase cannot be set if auth_type is password")
+        auth = "usePassword"
+    elif auth_type == "ssh_key":
+        if remote_password is not None:
+            aci.fail_json(msg="remote_password cannot be set if auth_type is ssh_key")
+        auth = "useSshKeyContents"
+    else:
+        auth = None
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fileRemotePath",
+            aci_rn="fabric/path-{0}".format(name),
+            module_object=name,
+            target_filter={"name": name},
+        ),
+        child_classes=["fileRsARemoteHostToEpg"]
+    )
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if management_epg is not None:
```

Comment:
can you add test case for management_epg?

Reply:
added test for management_epg